front/lib/resources/vault_resource.ts

```diff
@@ -159,14 +159,9 @@ export class VaultResource extends BaseResource<VaultModel> {
     return vaults.filter((vault) => vault.canList(auth));
   }
 
-  static async listWorkspaceVaultsAsAdmin(
-    auth: Authenticator
-  ): Promise<VaultResource[]> {
-    if (!auth.isAdmin()) {
-      return [];
-    }
-
-    return this.baseFetch(auth);
+  static async listWorkspaceVaultsAsMember(auth: Authenticator) {
+    const vaults = await this.baseFetch(auth);
+    return vaults.filter((vault) => vault.canList(auth) && vault.canRead(auth));
```

Comment:
You should not use both, I guess what you want is `vault.canRead()`. Might be worth a comment to explain that this is needed as we only want to display vaults where the user is a member in the Assistant  Builder and the settings.